providers/defltprov.c

```diff
@@ -552,15 +552,15 @@ static const OSSL_ALGORITHM deflt_keymgmt[] = {
     { PROV_NAMES_ML_KEM_1024, "provider=default", ossl_ml_kem_1024_keymgmt_functions,
       PROV_DESCS_ML_KEM_1024 },
 # if !defined(OPENSSL_NO_ECX)
-    { PROV_NAMES_X25519MLKEM768, "provider=default", ossl_mlx_x25519_kem_kmgmt_functions,
+    { PROV_NAMES_X25519MLKEM768, "provider=default,fips=yes", ossl_mlx_x25519_kem_kmgmt_functions,
```

Comment:
This is setting a terrible precedent.  Loading just the default provider (without FIPS provider) and asking for a FIPS algorithm should fail.  With this and the property query merging elsewhere, it work.  I.e. you've got yourself a non-FIPS validated algorithm despite asking for a FIPS one.

Reply:
NO! This algorithm won't work. The property query merging is not applied when internally fetching the ML-KEM-768. So yeah, this will be fetchable so as the decoders and encoders are but the implementation will error out.

Reply:
The TLS hybrid is not considered a cryptographic algorithm by FIPS either as it does a pure concatenation and nothing else.